packages/devtools_app/lib/src/inspector/inspector_tree.dart

```diff
@@ -810,32 +609,22 @@ abstract class InspectorTree {
 
   void animateToTargets(List<InspectorTreeNode> targets);
 
-  void onTap(Offset offset) {
-    final row = getRow(offset);
-    if (row == null) {
-      return;
-    }
+  void onExpandRow(InspectorTreeRow row) {
+    setState(() {
+      row.node.isExpanded = true;
+      if (config.onExpand != null) {
+        config.onExpand(row.node);
+      }
+    });
+  }
 
-    onTapIcon(row, row.node.renderObject?.hitTest(offset)?.icon);
+  void onCollapseRow(InspectorTreeRow row) {
+    setState(() {
+      row.node.isExpanded = false;
```

Comment:
do we not need a `config.onCollapse` call similar to what we do in `onExpandRow` above?

Reply:
No because the controller is not notified on row collapse but needs to be notified on row expand as expanding a row in the tree may trigger needing to load data from the server but collapsing a row does not.